arch/x86/mm/kasan_init_64.c

```diff
@@ -104,5 +194,33 @@ void __init kasan_init(void)
 		if (map_range(&pfn_mapped[i]))
 			panic("kasan: unable to allocate shadow!");
 	}
+	populate_zero_shadow(kasan_mem_to_shadow(PAGE_OFFSET + MAXMEM),
+			kasan_mem_to_shadow(KASAN_SHADOW_START));
+
+	populate_poison_shadow(kasan_mem_to_shadow(KASAN_SHADOW_START),
+			kasan_mem_to_shadow(KASAN_SHADOW_END));
+
+	if (!IS_ENABLED(CONFIG_KASAN_STACK)
+		&& !IS_ENABLED(CONFIG_KASAN_GLOBALS))
+		populate_zero_shadow(kasan_mem_to_shadow(KASAN_SHADOW_END),
+				KASAN_SHADOW_END);
+	else {
+		populate_zero_shadow(kasan_mem_to_shadow(KASAN_SHADOW_END),
+				kasan_mem_to_shadow(__START_KERNEL_map));
+		vmemmap_populate(kasan_mem_to_shadow((unsigned long)_stext),
+				kasan_mem_to_shadow((unsigned long)_end),
+				NUMA_NO_NODE);
+
+		if (!IS_ENABLED(CONFIG_KASAN_GLOBALS))
+			populate_zero_shadow(kasan_mem_to_shadow(MODULES_VADDR),
+					KASAN_SHADOW_END);
+		else
+			populate_zero_shadow(kasan_mem_to_shadow(MODULES_END),
+					KASAN_SHADOW_END);
+	}
+
+	memset(poisoned_page, 0xF9, PAGE_SIZE);
```

Comment:
Please either use another page as poisoned page, or comment on its declaration that it is used as 'scratch' shadow during early bootstrap when kasan is disabled, and later reused as poisoned page. I am not sure how it should be named in the latter case.